compose.yaml

```diff
@@ -47,3 +47,13 @@ services:
     command: bash -c "echo 'Copying node_modules, this may take a few minutes...' && rsync -rah --info=progress2 /node_modules /code/wagtail/ && npm run start"
     restart: "no"
     tty: true
+
+  elastic:
+    container_name: elastic
+    image: elasticsearch:8.17.1
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+    environment:
+      - xpack.security.enabled=false
+      - discovery.type=single-node
```

Comment:
This is to simplify accessing / running locally (single node, no security). I’ve copy-pasted this from existing projects, not sure how we arrived at those specific settings. It works.

Reply:
Yes, without `xpack.security.enabled=false` the Elasticsearch server will reject requests from the Django server process because Elasticsearch requires HTTPS by default. I ran into this while working on https://github.com/wagtail/wagtail/pull/12862 yesterday: https://stackoverflow.com/questions/71492404

I also use `discovery.type=single-node` per the example in https://hub.docker.com/_/elasticsearch, but I haven't tried if it works without that.

The Elasticsearch GitHub Action we use in Wagtail's CI has a few more parameters, including both we set here: https://github.com/getong/elasticsearch-action/blob/v1.3/entrypoint.sh#L7